src/client.rs

```diff
@@ -65,326 +63,251 @@ pub async fn run_client(
-							log::info!("block CID {:?} not present in local client", v.cid);
-						},
-					};
-				},
-				None => {
-					log::info!("error: no CID entry found for block {}", query.block);
-					if let Err(_) = query.res_chan.try_send(None) {
-						log::info!("error: failed to respond back to querying party");
-					}
-				},
-			}
-		}
+		cell_query_rx.into_iter().for_each(|query| {
+			let _ = do_cell_query(&cell_ipfs, &cell_store, query).inspect_err(into_warning);
+		});
 	});
 
-	// // block to CID mapping to locally kept here ( in thead safe manner ! )
+	// block to CID mapping to locally kept here ( in thead safe manner ! )
 
 	// broadcast fact i.e. some block number is mapped to some
 	// Cid ( of respective block data matrix ) to all peers subscribed
-	let mut fact_topic = ipfs.subscribe("topic/block_cid_fact").unwrap();
-	// ask over gossip network which Cid is associated with which block number
-	// and expect some one, who knows, will answer it
-	let mut ask_topic = ipfs.subscribe("topic/block_cid_ask").unwrap();
+	let fact_topic = ipfs.subscribe("topic/block_cid_fact").unwrap();
 
 	// listen for messages received on fact topic !
-	let block_cid_store_0 = block_cid_store.clone();
+	let fact_ipfs = ipfs.clone();
+	let fact_store = Arc::clone(&store);
 	tokio::task::spawn(async move {
-		loop {
-			let msg = fact_topic.next().await;
-			match msg {
-				Some(msg) => match msg {
-					ipfs_embed::GossipEvent::Subscribed(peer) => {
-						log::info!("subscribed to topic `topic/block_cid_fact`\t{}", peer);
-					},
-					ipfs_embed::GossipEvent::Unsubscribed(peer) => {
-						log::info!("unsubscribed from topic `topic/block_cid_fact`\t{}", peer);
-					},
-					ipfs_embed::GossipEvent::Message(peer, msg) => {
-						if let Some((block, cid)) = decode_block_cid_fact_message(msg.to_vec()) {
-							{
-								match get_block_cid_entry(block_cid_store_0.clone(), block) {
-									Some(v) => {
-										if v.self_computed && v.cid != cid {
-											log::info!(
-												"received CID doesn't match host computed CID"
-											);
-										}
-										// @note what happens if have-CID is not host computed and
-										// CID mismatch is encountered ?
-										//
-										// Need to verify/ self-compute CID and reach to a (more) stable
-										// state
-									},
-									None => {
-										match set_block_cid_entry(
-											block_cid_store_0.clone(),
-											block,
-											BlockCidPair {
-												cid,
-												self_computed: false, // because this block CID is received over network !
-											},
-										) {
-											Ok(_) => {},
-											Err(e) => {
-												log::info!("error: {}", e);
-											},
-										};
-									},
-								};
-							}
-
-							log::info!("received message on `topic/block_cid_fact`\t{}", peer);
-						}
-
-						// received message was not decodable !
-					},
-				},
-				None => {
-					break;
-				},
-			}
-		}
+		fact_topic.for_each(|msg| {
```

Comment:
Not sure why we break out of a loop if next is None, and why it could be none, but just to be sure, does for_each keeps same meaning?